src/query/list_file.c

```diff
@@ -154,20 +151,21 @@ struct qfile_list_cache_entry_pool
   QFILE_POOLED_LIST_CACHE_ENTRY *pool;	/* pre-allocated array */
   int n_entries;		/* number of entries in the pool */
   int free_list;		/* the head(first entry) of the free list */
+  pthread_mutex_t pool_mutex;	/* for thread safe */
```

Comment:
This mutex seems to manage the qfile_List_cache_entry_pool
- get an entry from the entry pool
- put newly allocated entry to entry pool

We seem have a lock-free concept data structures in base/lockfree::*
lockfree::circular_queue could be used to avoid the cost of mutex locking.

Reply:
I found the mutex **pool_mutex** is not needed to control the pool because the rountine **qfile_allocate_list_cache_entry** and qfile_free_list_cache_entry is called with critical section **CSECT_QPROC_LIST_CACHE** from the routines related to cache entry managing. So, I will remove the mutex.